CONTRIBUTING.md

```diff
@@ -0,0 +1,190 @@
+# Contributing
+
+This document provides guidelines for contributing to this repository.
+
+## Development Environment Setup
+
+### Prerequisites
+* Basic tools: `curl`, `jq`, `oc`
+* go: v1.19
+* Container engine: `podman`
+
+## Running a test environment
+
+### Kwok Container
+"Kwok" is a Kubernetes SIGs-hosted project. KWOK is an abbreviation for Kubernetes Without Kubelet. Kwok simply simulates the node's behaviour. As a result, it can mimic a high number of nodes and pods while consuming only a small amount of memory.
+
+To run the Kwok container with the Kwok Kubernetes tool, follow these steps:
+
+1. Bring up the Kwok cluster by running the following command:
+
+```
+podman run --rm -d -p 8080:8080 --name kwok-cluster registry.k8s.io/kwok/cluster:v0.3.0-k8s.v1.27.3
+```
+
+2. Once the Kwok cluster is up and running, set the cluster details in the OpenShift client with the following command:
+
+```
+oc config set-cluster kwok --server=http://127.0.0.1:8080
+```
+
+3. Create a new context (you only need to set the cluster once) for the Kwok cluster with the following command:
+
+```
+oc config set-context kwok --cluster=kwok
+```
+4. Set the Kwok context as the current context, if you've previously switched to another cluster, with the following command:
+```
+oc config use-context kwok
+```
+
+Now you can access the cluster using kubectl, e.g.: `kubectl get ns`.
+
+### Setting up a containerized Splunk instance
+
+To set up a containerized Splunk instance, you can use either podman or docker.
+Follow these steps:
+
+1. Run the container with the desired configuration:
+   ```
+   podman run -p 8000:8000 -p 8089:8089 -e "SPLUNK_PASSWORD=YourPassword" -e "SPLUNK_START_ARGS=--accept-license" -it --name splunk_container docker.io/splunk/splunk:latest
+   ```
+    This command maps port 8000 and port 8089 from the container to the host machine, 
+    set a password that is a minimum of 8 characters long (the minimum requirement),
+    and accepts the Splunk license.
+    For more information about the specific command inputs and options, refer
+    to the [Splunk documentation][CS1].
+2. Once the container is running, you can log in to the Splunk instance using
+   the username `admin` and the password you set up.
+3. To access the Splunk [REST API][CS2],
+   you can make API calls from outside the container using the `curl` command.
+   For example, you can run the following command to search all data:
+     ```
+     curl -u admin:YourPassword -k https://localhost:8089/services/search/jobs -d search="search *"
+     ```
+4. If you want to use the Splunk UI, open a web browser on the host and navigate to
```

Comment:
There seems to be an unclosed quote somewhere, because syntax highlighting seems broken from this point on